bin/ch-completion.bash

```diff
@@ -0,0 +1,388 @@
+
+    # If "$cur" is non-empty, we want to ignore it as a potential subcommand
+    # to avoid unwanted behavior. “${words[@]::${#words[@]}-1}” gives you all
+    # but the last element of the array “words” (see syntax glossary above) so 
+    # if "$cur" is non-empty, pass it to _ch_subcommand_get. Otherwise pass all
+    # elements of “words.”
+    if [[ -n "$cur" ]]; then
+        sub_cmd=$(_ch_subcommand_get "$_image_subcommands" "${words[@]::${#words[@]}-1}")
+        strg_dir=$(_ch_find_storage "${words[@]::${#words[@]}-1}")
+    else
+        sub_cmd=$(_ch_subcommand_get "$_image_subcommands" "${words[@]}")
+        strg_dir=$(_ch_find_storage "${words[@]}")
+    fi
+
+    # Common opts that take args
+    #
+    case "$prev" in
+    -a | --arch )
+        # FIXME: Remove yolo?
+        # FIXME: Missing common architectures?
+        COMPREPLY=( $(compgen -W "host yolo 386 amd64 arm/v6 
+                                  arm/v7 arm64/v8 ppc64le s390x" -- $cur) )
+        return 0
+        ;;
+    --cache-large )
+        # This is just a user-specified number. Can't autocomplete
+        COMPREPLY=()
+        return 0
+        ;;
+    -s | --storage )
+        # This “if” helps avoid overzealous completion. E.g. if there's
+        # only one subdir of the current dir, this command completes to
+        # that dir even if "$cur" is empty. I didn't like that, hence the
+        # “if”.
+        if [[ -n "$cur" ]]; then
+            compopt -o nospace
+            COMPREPLY=( $(compgen -d -S / -- $cur) )
+        fi
+        return 0
+        ;;
+    esac
+
+    case "$sub_cmd" in
+    build )
+        # FIXME: opts with args that I need to revisit for this logic
+        #
+        #   * --build-arg KEY[=VALUE]
+        #   * --force-cmd=CMD,ARG1[,ARG2...]
+        #
+        case "$prev" in
+        # Go through a list of potential subcommand-specific opts to see if
+        # “$cur” should be an argument. Otherwise, default to CONTEXT or any
+        # valid option (common or subcommand-specific).
+        -f|--file )
+            compopt -o nospace
+            COMPREPLY=( $(_compgen_filepaths "$cur") )
+            return 0
+            ;;
+        -t )
+            # We can't autocomplete a tag, so we're not even gonna allow
+            # autocomplete after this option.
+            COMPREPLY=()
+            return 0
+            ;;
+        *)
+            # Autocomplete to context directory, common opt, or build-specific opt
+            echo "compgen bit:" >> /tmp/ch-completion.log
+            # --force can take “fakeroot” or “seccomp” as an argument, or no
+            # argument at all. To account for this, we add those two arguments
+            # to the list of compgen suggestions, which allows compgen to autocomplete
+            # to “fakeroot,” “seccomp,” or anything that could logically follow
+            # “--force” with no argument.
+            if [[ "$prev" == "--force" ]]; then
```

Comment:
Does this deal with equals-separated options too? E.g. `--force=seccomp` is also valid.